plugins/modules/ipa_getkeytab.py

```diff
@@ -0,0 +1,250 @@
+      - This is incompatible with the O(password), and will work only against a IPA server more recent than version 3.3.
+      - The user requesting the keytab must have access to the keys for this operation to succeed.
+      - Be aware that if set V(true), a new keytab will be generated.
+      - This invalidates all previously retrieved keytabs for this service principal.
+    type: bool
+  encryption_types:
+    description:
+      - The list of encryption types to use to generate keys.
+      - It will use local client defaults if not provided.
+      - Valid values depend on the Kerberos library version and configuration.
+    type: str
+  state:
+    description:
+      - The state of the keytab file.
+      - V(present) only check for existence of a file, if you want to recreate keytab with other parameters you should set O(force=true).
+    type: str
+    default: present
+    choices: ["present", "absent"]
+  force:
+    description:
+      - Force recreation if exists already.
+    type: bool
+requirements:
+  - freeipa-client
+  - Managed host is FreeIPA client
+extends_documentation_fragment:
+  - community.general.attributes
+'''
+
+EXAMPLES = r'''
+- name: Get kerberos ticket
+  ansible.builtin.shell: kinit admin
+  args:
+    stdin: "{{ aldpro_admin_password }}"
+  changed_when: true
+
+- name: Create keytab
+  community.general.ipa_getkeytab:
+    path: /etc/ipa/test.keytab
+    principal: HTTP/freeipa-dc02.ipa.test
+    ipa_server: freeipa-dc01.ipa.test
+
+- name: Retrieve already existing keytab
+  community.general.ipa_getkeytab:
+    path: /etc/ipa/test.keytab
+    principal: HTTP/freeipa-dc02.ipa.test
+    ipa_server: freeipa-dc01.ipa.test
+    retrieve_mode: true
+
+- name: Force keytab recreation
+  community.general.ipa_getkeytab:
+    path: /etc/ipa/test.keytab
+    principal: HTTP/freeipa-dc02.ipa.test
+    ipa_server: freeipa-dc01.ipa.test
+    force: true
+'''
+
+import os
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.community.general.plugins.module_utils.cmd_runner import CmdRunner, cmd_runner_fmt
+
+
+class IPAKeytab(object):
+    def __init__(self, module, **kwargs):
+        self.module = module
+        self.path = kwargs['path']
+        self.state = kwargs['state']
+        self.principal = kwargs['principal']
+        self.ipa_server = kwargs['ipa_server']
+        self.ldap_uri = kwargs['ldap_uri']
+        self.bind_dn = kwargs['bind_dn']
+        self.bind_pw = kwargs['bind_pw']
+        self.password = kwargs['password']
+        self.ca_certificate = kwargs['ca_certificate']
+        self.sasl_mech = kwargs['sasl_mech']
+        self.retrieve_mode = kwargs['retrieve_mode']
+        self.encryption_types = kwargs['encryption_types']
+
+        self.executable = [module.get_bin_path('ipa-getkeytab', True)]
+        self.runner = CmdRunner(
+            module,
+            command=self.executable,
```

Comment:
This could be made simpler with:
```suggestion
        self.runner = CmdRunner(
            module,
            command='ipa-getkeytab',
```